app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -234,6 +279,47 @@ class _CustomChatInputState extends ConsumerState<CustomChatInput> {
                       ),
                     ),
                   ),
+                  if(showEditButton)
+                    InkWell(
+                      onTap: () async {
+                        final emojiRowVisible = ref.read(
+                          chatInputProvider(roomId).select((ci) {
+                            return ci.emojiRowVisible;
+                          }),
+                        );
+                        final inputNotifier =
+                        ref.read(chatInputProvider(roomId).notifier);
+                        if (emojiRowVisible) {
+                          inputNotifier.setCurrentMessageId(null);
+                          inputNotifier.emojiRowVisible(false);
+                        }
+
+                        inputNotifier.toggleEditView(true);
+                        final message = ref
+                            .read(chatStateProvider(widget.convo))
+                            .messages
+                            .firstWhere(
+                              (element) => element.id == currentMessageId,
+                        );
+                        chatInputNotifier.setEditMessage(message);
+                        if (message is TextMessage) {
+                          ref
+                              .read(_textValuesProvider(roomId).notifier)
+                              .update((state) => message.text);
+                        }
+
+                        final chatInputFocusState =
+                        ref.read(chatInputFocusProvider.notifier);
+                        WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+                          FocusScope.of(context)
+                              .requestFocus(chatInputFocusState.state);
+                        });
+                      },
+                      child: const Text(
+                        'Edit',
+                        style: TextStyle(color: Colors.white),
```

Comment:
Avoid hardcoding theme variables.

Reply:
Actually, I have followed similar code that was already there for other menu options label as below.

 ```
InkWell(
                    onTap: () => customMsgSnackbar(
                      context,
                      'More options not implemented yet',
                    ),
                    child: const Text(
                      'More',
                      style: TextStyle(color: Colors.white),
                    ),
                  ),
```

Reply:
Ah, I just removed `style: TextStyle(color: Colors.white),` as default textTheme have white color.